src/routes/leases.ts

```diff
@@ -1,48 +1,58 @@
-import express, { Request, Response } from 'express';
-import { getTenantByUserId } from '../db/models/tenant';
-import { updateLease } from '../db/models/lease';
+import express, { Request, Response } from "express";
+import { getTenantByUserId } from "../db/models/tenant";
+import { updateLease } from "../db/models/lease";
 
 const router = express.Router();
 
-router.get('/view', async (req: Request, res: Response) => {
-    const userId = req.user.userId;
-    const tenant = await getTenantByUserId(userId);
+router.get("/view", async (req: Request, res: Response) => {
+  const userId = req.user.userId;
+  const tenant = await getTenantByUserId(userId);
 
-    if (!tenant) {
-        res.status(404).json({ message: 'Tenant not found' });
-        return;
-    };
+  if (!tenant) {
+    res.status(404).json({ message: "Tenant not found" });
+    return;
+  } else if (
+    !tenant.unitTenants?.length ||
+    !tenant.unitTenants[0].leases?.length
+  ) {
+    res.status(200).json({ message: "No leases found." });
+    return;
+  }
 
-    const leases = tenant.unitTenants[0].leases;
-    const renewalLease = leases.filter(lease => {
-        return lease.status === 'pending';
-    });
+  const leases = tenant.unitTenants[0].leases;
+  const renewalLease = leases.filter((lease) => {
+    return lease.status === "pending";
+  });
 
-    if (!renewalLease.length) {
-        res.status(200).json({ message: 'No pending renewals.' });
-        return;
-    }
+  if (!renewalLease.length) {
+    res.status(200).json({ message: "No pending renewals." });
+    return;
+  }
 
-    const tenantLeaseDetails = {
-        ...tenant,
-        ...renewalLease[0]
-    };
+  const tenantLeaseDetails = {
+    ...tenant,
+    ...renewalLease[0],
+  };
 
```

Comment:
suggestion(non-blocking): this is fine, but can we refactor to reduce nesting and improve readability?